### ISSUE TITLE
Jupyter web app 500 error listing poddefaults

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#3918]
/kind bug

**What steps did you take and what happened:**

I'm trying out v0.6.2 RC which is using
gcr.io/kubeflow-images-public/jupyter-web-app:e8a8a6f

When I open up the new server window I see an error

```
main.d8acb8b4497ff95ec300.js:1 ERROR Http failure response for https://jlewi-0-6-2-rc.endpoints.jlewi-dev.cloud.goog/jupyter/api/namespaces/kubeflow-jlewi/poddefaults: 500 OK
```

So its not able to list poddefaults in my namespace

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard):
- kfctl version: (use `kfctl version`):
- Kubernetes platform: (e.g. `minikube`)
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by jlewi** [comment]
Here's the deployment

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: 2019-08-16T04:09:32Z
  generation: 1
  labels:
    app: jupyter-web-app
    app.kubernetes.io/component: jupyter
    app.kubernetes.io/instance: jupyter-web-app
    app.kubernetes.io/managed-by: kfctl
    app.kubernetes.io/name: jupyter-web-app
    app.kubernetes.io/part-of: kubeflow
    app.kubernetes.io/version: v0.6
    kustomize.component: jupyter-web-app
  name: jupyter-web-app-deployment
  namespace: kubeflow
  ownerReferences:
  - apiVersion: app.k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: false
    kind: Application
    name: jupyter-web-app
    uid: a6f0cb71-bfdb-11e9-82c7-42010a8e010d
  resourceVersion: "2860"
  selfLink: /apis/extensions/v1beta1/namespaces/kubeflow/deployments/jupyter-web-app-deployment
  uid: a6932030-bfdb-11e9-82c7-42010a8e010d
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: jupyter-web-app
      app.kubernetes.io/component: jupyter
      app.kubernetes.io/instance: jupyter-web-app
      app.kubernetes.io/managed-by: kfctl
      app.kubernetes.io/name: jupyter-web-app
      app.kubernetes.io/part-of: kubeflow
      app.kubernetes.io/version: v0.6
      kustomize.component: jupyter-web-app
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: jupyter-web-app
        app.kubernetes.io/component: jupyter
        app.kubernetes.io/instance: jupyter-web-app
        app.kubernetes.io/managed-by: kfctl
        app.kubernetes.io/name: jupyter-web-app
        app.kubernetes.io/part-of: kubeflow
        app.kubernetes.io/version: v0.6
        kustomize.component: jupyter-web-app
    spec:
      containers:
      - env:
        - name: ROK_SECRET_NAME
          valueFrom:
            configMapKeyRef:
              key: ROK_SECRET_NAME
              name: jupyter-web-app-parameters
        - name: UI
          valueFrom:
            configMapKeyRef:
              key: UI
              name: jupyter-web-app-parameters
        image: gcr.io/kubeflow-images-public/jupyter-web-app:e8a8a6f
        imagePullPolicy: Always
        name: jupyter-web-app
        ports:
        - containerPort: 5000
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /etc/config
          name: config-volume
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: jupyter-web-app-service-account
      serviceAccountName: jupyter-web-app-service-account
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: jupyter-web-app-config
        name: config-volume

```

Here's the cluster role

```
kubectl -n kubeflow get -o yaml clusterrole -o yaml jupyter-web-app-cluster-role
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: 2019-08-16T04:09:28Z
  labels:
    app: jupyter-web-app
    app.kubernetes.io/component: jupyter
    app.kubernetes.io/instance: jupyter-web-app
    app.kubernetes.io/managed-by: kfctl
    app.kubernetes.io/name: jupyter-web-app
    app.kubernetes.io/part-of: kubeflow
    app.kubernetes.io/version: v0.6
    kustomize.component: jupyter-web-app
  name: jupyter-web-app-cluster-role
  resourceVersion: "2659"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/jupyter-web-app-cluster-role
  uid: a46e99be-bfdb-11e9-82c7-42010a8e010d
rules:
- apiGroups:
  - ""
  resources:
  - namespaces
  verbs:
  - get
  - list
  - create
  - delete
- apiGroups:
  - kubeflow.org
  resources:
  - notebooks
  - poddefaults
  verbs:
  - get
  - list
  - create
  - delete
- apiGroups:
  - ""
  resources:
  - persistentvolumeclaims
  verbs:
  - create
  - delete
  - get
  - list
- apiGroups:
  - storage.k8s.io
  resources:
  - storageclasses
  verbs:
  - get
  - list
  - watch
```

So looks like we should have poddefaults permission.

**Comment by jlewi** [comment]
Here's the error

```
[2019-08-16 05:49:14,829] ERROR in app: Exception on /api/namespaces/kubeflow-jlewi/poddefaults [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/dist-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/app/kubeflow_jupyter/common/base_app.py", line 45, in get_poddefaults
    desc = pd["spec"]["desc"]
KeyError: 'desc'
10.52.0.25 - - [16/Aug/2019 05:49:14] "GET /api/namespaces/kubeflow-jlewi/poddefaults HTTP/1.1" 500 -
10.52.1.17 - - [16/Aug/2019 05:49:14] "GET /api/namespaces/kubeflow-jlewi/pvcs HTTP/1.1" 200 -
10.52.1.7 - - [16/Aug/2019 05:49:14] "GET /api/namespaces/kubeflow-jlewi/notebooks HTTP/1.1" 200 -
```
It looks like the problem is that the description is empty.

This was fixed in #3845

It was committed to kubeflow/manifests in kubeflow/manifests#269

But not cherry-picked to kubeflow/manifests/v0.6.2

**Comment by jlewi** [comment]
h

**Comment by jlewi** [comment]
Cherry-pick is kubeflow/manifests#280

**Comment by jlewi** [comment]
Here's the pod default

```
apiVersion: kubeflow.org/v1alpha1
desc: add gcp credential
env:
- name: GOOGLE_APPLICATION_CREDENTIALS
  value: /secret/gcp/user-gcp-sa.json
kind: PodDefault
metadata:
  creationTimestamp: 2019-08-16T04:13:51Z
  generation: 1
  name: add-gcp-secret
  namespace: kubeflow-jlewi
  resourceVersion: "4861"
  selfLink: /apis/kubeflow.org/v1alpha1/namespaces/kubeflow-jlewi/poddefaults/add-gcp-secret
  uid: 40d741a2-bfdc-11e9-82c7-42010a8e010d
spec:
  selector:
    matchLabels:
      add-gcp-secret: "true"
volumeMounts:
- mountPath: /secret/gcp
  name: secret-volume
volumes:
- name: secret-volume
  secret:
    secretName: user-gcp-sa
```

It looks like the desc field is set incorrectly; its not in the spec.
https://github.com/kubeflow/kubeflow/blob/b0d2e1a68cb2e9fcb874dfbcbae5ac1f560e1366/bootstrap/pkg/kfapp/gcp/gcp.go#L1691

**Comment by jlewi** [comment]
So to summarize there were two issues here

1. The Default PodDefault created by kfctl was incorrect and didn't include a description
    * This is #3924 and should be fixed
1. The jupyter web app didn't handle the case of PodDefaults without descriptions
    * This is fixed by kubeflow/manifests#280

I'm going to mark this as fixed. We still need to verify but I think we can do that as part of getting the #3860 RC ready.

**Comment by jlewi** [comment]
Verified; see http://bit.ly/2ZgOE1v 
that I no longer get 500 on GCP with IAP.
The default PodDefaults has been fixed.
Jupyter web app should be using the fixed image that handles missing description but I didn't verify that.
